src/interfaces/docker_interface.py

```diff
@@ -158,25 +131,28 @@ def get_messages(self, a_name: str, b_ipv4: str, bitcoin_network: str = "regtest
         messages.sort(key=lambda x: x["time"])
         return messages
 
-
-    def logs_grep(self, pattern: str, container_name: str):
+    def logs_grep(self, pattern: str, container_name: str) -> str:
         compiled_pattern = re.compile(pattern)
-
-        now = datetime.utcnow()
-        log_stream = self.client.api.logs(
-            container=container_name,
-            stdout=True,
-            stderr=True,
-            stream=True,
-            until=now,
-        )
-
+        command = f"docker logs {container_name}"
+        args = shlex.split(command)
         matching_logs = []
-        for log_entry in log_stream:
-            log_entry_str = log_entry.decode('utf-8').strip()
-            if compiled_pattern.search(log_entry_str):
-                matching_logs.append(log_entry_str)
 
+        try:
+            with subprocess.Popen(
```

Comment:
why didn't you use `run_subprocess()` here?

Reply:
oh I see, because we're filtering results with `grep`

Reply:
Sorry yeah. That's why. I think clever code could still merge the two, but figured it was an optimisation